### PR TITLE
Ignore private security forks in label_sync

### LINKED ISSUE #15157: label_sync is failing
https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-test-infra-label-sync/1192264870515445760

https://k8s-testgrid.appspot.com/sig-testing-misc#label_sync

from some of the logs
> {"component":"label_sync","file":"label_sync/main.go:358","func":"main.loadLabels.func1","level":"error","msg":"Failed listing labels for repo","org":"kubernetes-sigs","repo":"krew-ghsa-m9g4-7496-9p6f","time":"2019-11-07T01:18:11Z"}

> {"component":"label_sync","file":"label_sync/main.go:358","func":"main.loadLabels.func1","level":"error","msg":"Failed listing labels for repo","org":"kubernetes-sigs","repo":"krew-ghsa-m9g4-7496-9p6f","time":"2019-11-07T02:18:12Z"}

### PR DESCRIPTION
**Pull request by cblecker**
We can't modify labels on private security forks, so filter them out.

/assign @BenTheElder 

fixes #15157

### CODE REVIEW

File: label_sync/main.go [hunk 1]
```diff
@@ -319,6 +319,10 @@ func GetOrg(org string) (string, bool) {
 	return org, false
 }
 
+// securityForkNameRE is a regexp matching repos that are temporary security forks
+// https://help.github.com/en/github/managing-security-vulnerabilities/collaborating-in-a-temporary-private-fork-to-resolve-a-security-vulnerability
+var securityForkNameRE = regexp.MustCompile(`^[\w-]+-ghsa-[\w-]+$`)
```
Comment (BenTheElder) [inline label_sync/main.go]: it's a little unfortunate that we have to ignore some magical pattern :/

we should probably document this, however I'd prefer to get label_sync _working_ and have someone follow up

**Comment by BenTheElder** [comment]
FYI @cjwagner @stevekuznetsov @alvaroaleman 
/lgtm
/approve

File: label_sync/main.go (hunk 1 above)
Comment (cblecker) [reply label_sync/main.go]: I couldn't see any extra flag in the v3 API that calls out that this is a private security fork.

Comment (BenTheElder) [reply label_sync/main.go]: same. IMO this is a reasonable fix to an unreasonable problem to have. It's not great that any repo with this string will magically be ignored but :man_shrugging:

Comment (BenTheElder) [reply label_sync/main.go]: feedback for github really

**Comment by cjwagner** [comment]
Did we reach out to GH support about this yet?

**Comment by BenTheElder** [comment]
I didn't but I also couldn't see much about this from my end.

**Pull request merged**